### PR TITLE
optimize ERC721 inventory query

### PR DESCRIPTION
**Pull request by ayrat555**
While working on https://github.com/poanetwork/blockscout/issues/2541 I found :

Inventory tab for ERC721 doesn't load
any token transfers because query is timing
out.

Looking into the query, it does a couple of
unnecessary joins:
1. with `transactions` for `block_number`. But
token_transfers already have block_number
2. with `tokens` to check if `token_transfers` are
for ERC-721 token. But the query is only executed only
for ERC 721 tokens

One more issue that I found is that the query uses distinct
by `token_id`. But we don't have any indexes on `token_id`.
So this PR adds an index on `token_id`.

## Changelog

- optimize ERC721 inventory query

### CODE REVIEW

File: apps/explorer/test/explorer/chain/token_transfer_test.exs [hunk 1]
```diff
@@ -188,31 +188,6 @@ defmodule Explorer.Chain.TokenTransferTest do
       assert Enum.map(results, & &1.token_id) == [last_owner.token_id]
       assert Enum.map(results, & &1.to_address_hash) == [last_owner.to_address_hash]
     end
-
-    test "won't return tokens that aren't uniques" do
```
Comment (pasqu4le) [inline apps/explorer/test/explorer/chain/token_transfer_test.exs]: Why did you remove this test?

Comment (ayrat555) [reply apps/explorer/test/explorer/chain/token_transfer_test.exs]: because it checks if token is `ERC-20`. But the query is executed only for `ERC-721` tokens
